libp2p/protocols/secure/noise.nim

```diff
@@ -0,0 +1,476 @@
+## Nim-LibP2P
+## Copyright (c) 2020 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import chronos
+import chronicles
+import nimcrypto/[utils, sysrand, sha2, hmac]
+import ../../connection
+import ../../peer
+import ../../protobuf/minprotobuf
+import secure,
+       ../../crypto/[crypto, chacha20poly1305, curve25519, hkdf]
+
+template unimplemented: untyped =
+  doAssert(false, "Not implemented")
+
+const
+  # https://godoc.org/github.com/libp2p/go-libp2p-noise#pkg-constants
+  NoiseCodec* = "/noise"
```

Comment:
is this the proper noise codec? Seems a bit non-standard for libp2p.

Reply:
Well see the link in the comment.. I had to dig it actually! there is no mention in the spec I could find 😄